Engine/Source/GameObject.h

```diff
@@ -92,6 +101,7 @@ class GameObject
 	std::vector<Component*> mComponents;
 	std::vector<Component*> mComponentsToDelete;
 	const unsigned int mID;
+	unsigned int mParentID;
```

Comment:
I think a game objects should have a Parent game object pointr (mParent) or the parentId (mParentID) but I think it does not have sense to have both of them stored. I think is easier to work with the GameObject*.